Tchou-Tchou/WifiAPI.swift

```diff
@@ -1,64 +1,82 @@
 import Cocoa
 
 class WifiAPI: NSObject {
-    var baseURL: String = "https://wifi.sncf"
-
+    var supportedSystems = [
+        [
+            "ssid" : "_SNCF_WIFI_INOUI",
+            "base" : "https://wifi.sncf",
+            "endpoint" : "/router/api/train/gps"
+        ],
+        [
+            "ssid" : "WIFIonICE",
+            "base" : "https://iceportal.de/",
+            "endpoint" : "/api1/rs/status"
+        ]
+    ]
+    
     override init() {
         if let baseURL = UserDefaults.standard.string(forKey: "DebugWifiAPIBaseURL") {
-            self.baseURL = baseURL
+            self.supportedSystems[0]["base"] = baseURL
         }
     }
-
-    func fetchSpeed(completion: ((Double?)->(Void))?) {
-        let url = URL(string: baseURL + "/router/api/train/gps")!
+    
+    func fetchSpeed(system: String, completion: ((Double?)->(Void))?) {
+        var currentSystem = supportedSystems[0]
+        for (supportedSystem) in supportedSystems {
```

Comment:
`first(where:)` would be a great way to simplify this code :)

https://developer.apple.com/documentation/swift/dictionary/1848649-first